backend/src/apiserver/model/resource_reference.go

```diff
@@ -43,3 +57,13 @@ type ResourceReference struct {
 	// The json formatted blob of the resource reference.
 	Payload string `gorm:"column:Payload; not null; size:65535 "`
 }
+
+type ReferenceKey struct {
+	Type ResourceType
+	ID   string
+}
+
+type FilterContext struct {
```

Comment:
We don't create a DB table for `FilterContext`, so I think it's more reasonable for it to stay in `common` folder. Same for other types moved here.

Reply:
As we decided to remove the dependency on the api packages, we needed to have an internal representation of the types. Using the `model` enforces compatibility with the DB schema.